internal/controller/operator/deployment.go

```diff
@@ -172,6 +183,70 @@ func (r *AuthenticationReconciler) removeCP2Deployments(ctx context.Context, req
 	return subreconciler.ContinueReconciling()
 }
 
+// getAuditSecretNameIfExists determines whether an audit service has been
+// configured with TLS. Returns the name of the Secret used to store the TLS
+// certificates if a Secret has been identified by the user and found on the
+// cluster, or an empty string when the Secret isn't found or cannot otherwise
+// be retrieved. If an error other than NotFound is received when trying to get
+// the Secret, that is returned as well.
+func (r *AuthenticationReconciler) getAuditSecretNameIfExists(ctx context.Context, namespace string) (string, error) {
```

Comment:
You should just check the Authentication CR for whether the audit values are set - that's ultimately the source of truth for the workload and makes this unnecessary.

Reply:
ok I will change this to check the Authentication CR